File: UdidServer/src/horizon/util/StringUtil.java
package horizon.util;

public class StringUtil {
    public static String removeChar(String src, char ch){
        if(src == null || src.isEmpty()){
            return src;
        }
        if(src.indexOf(ch) < 0){
            return src;
        }

        char[] a  = src.toCharArray();
        int len = a.length;
        int p = 0;
        for (int i = 0; i < len; i++) {
            if(a[i] != ch){
                a[p++] = a[i];
            }
        }
        return new String(a, 0, p);
    }
}

File: UdidServer/src/main/kotlin/horizon/util/StringUtil.kt
package horizon.util

object StringUtil {
    fun removeChar(src: String?, ch: Char): String? {
        if (src == null || src.isEmpty()) {
            return src
        }
        if (src.indexOf(ch) < 0) {
            return src
        }
        val a = src.toCharArray()
        val len = a.size
        var p = 0
        for (i in 0 until len) {
            if (a[i] != ch) {
                a[p++] = a[i]
            }
        }
        return String(a, 0, p)
    }
}
